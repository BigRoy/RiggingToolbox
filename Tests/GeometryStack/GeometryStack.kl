
require RiggingToolbox;

operator entry(){
   	
   	GeometryStack stack();
   	PolygonMeshSphereGenerator sphereGenerator(2.0, 1, true, true);
   	PushDeformer pushDeformer(3.0);
   	sphereGenerator.enableCaching = true;
   	stack.addGeometryOperator(sphereGenerator);
   	stack.addGeometryOperator(pushDeformer);
   	stack.initCachePoints();
   	stack.disableCachePoint(1);

    EvalContext context();
    {
	    GeometrySet geomSet = stack.evaluate(context);

	    report("NumGeometries:" + geomSet.size);
	    for(Integer i=0; i<geomSet.size; i++){
	        report(geomSet.get(i).getDesc(true, true));
	    }
	}
	
	{
	    GeometrySet geomSet = stack.evaluate(context);

	    report("NumGeometries:" + geomSet.size);
	    for(Integer i=0; i<geomSet.size; i++){
	        report(geomSet.get(i).getDesc(true, true));
	    }
	}
}

