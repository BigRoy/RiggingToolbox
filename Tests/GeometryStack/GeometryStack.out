NumGeometries:1
Mesh: pointCount: 12 polygonCount: 15 nbAttributeVectors: 24
  Points (adjacent polygons as 'polygon.neighborPolygonIndex', 
          borders as '|', closed wing starts as '<<',
          unshared attribute as value@attributeIndex ):
    0: 5 polygons:  <<4.0, 0.0, 1.0, 2.0, 3.0
      Attr positions:{x:+0.0,y:+5.0,z:+0.0}
      Attr normals:{x:+0.0,y:+1.0,z:+0.0}
      Attr uvs0:{x:+0.5,y:+0.0}
    1: 4 polygons:  |0.1, 4.2,  |5.0, 9.1
      Attr positions:{x:+4.330078,y:+2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:+0.5,z:+0.0}
      Attr uvs0:{x:+0.800048,y:+0.0}@1 {x:+0.200012,y:+0.333313}@16 {x:+1.0,y:+0.333313}@17 {x:+0.0,y:+0.333313}@22 
    2: 4 polygons:  |1.1, 0.2,  |6.0, 5.1
      Attr positions:{x:+1.338134,y:+2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:+0.823608}
      Attr uvs0:{x:+0.599975,y:+0.0}@12 {x:+1.0,y:+0.333313}@2 {x:+0.800048,y:+0.333313}@18 {x:+0.800048,y:+0.333313}@18 
    3: 4 polygons:  |2.1, 1.2,  |7.0, 6.1
      Attr positions:{x:-3.502929,y:+2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:+0.509033}
      Attr uvs0:{x:+0.400024,y:+0.0}@13 {x:+0.800048,y:+0.333313}@3 {x:+0.599975,y:+0.333313}@19 {x:+0.599975,y:+0.333313}@19 
    4: 4 polygons:  |3.1, 2.2,  |8.0, 7.1
      Attr positions:{x:-3.502929,y:+2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:-0.509033}
      Attr uvs0:{x:+0.200012,y:+0.0}@14 {x:+0.599975,y:+0.333313}@4 {x:+0.400024,y:+0.333313}@20 {x:+0.400024,y:+0.333313}@20 
    5: 4 polygons:  |4.1, 3.2,  |9.0, 8.1
      Attr positions:{x:+1.338134,y:+2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:-0.823608}
      Attr uvs0:{x:+0.0,y:+0.0}@15 {x:+0.400024,y:+0.333313}@5 {x:+0.200012,y:+0.333313}@21 {x:+0.200012,y:+0.333313}@21 
    6: 4 polygons:  <<14.1, 9.2, 5.3, 10.0
      Attr positions:{x:+4.330078,y:-2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:-0.5,z:+0.0}
      Attr uvs0:{x:+0.0,y:+0.666626}@23 {x:+0.0,y:+0.666626}@23 {x:+1.0,y:+0.666626}@6 {x:+1.0,y:+0.666626}@6 
    7: 4 polygons:  <<11.0, 10.1, 5.2, 6.3
      Attr positions:{x:+1.338134,y:-2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:+0.823608}
      Attr uvs0:{x:+0.800048,y:+0.666626}
    8: 4 polygons:  <<12.0, 11.1, 6.2, 7.3
      Attr positions:{x:-3.502929,y:-2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:+0.509033}
      Attr uvs0:{x:+0.599975,y:+0.666626}
    9: 4 polygons:  <<13.0, 12.1, 7.2, 8.3
      Attr positions:{x:-3.502929,y:-2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:-0.509033}
      Attr uvs0:{x:+0.400024,y:+0.666626}
    10: 4 polygons:  <<14.0, 13.1, 8.2, 9.3
      Attr positions:{x:+1.338134,y:-2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:-0.823608}
      Attr uvs0:{x:+0.200012,y:+0.666626}
    11: 5 polygons:  <<14.2, 10.2, 11.2, 12.2, 13.2
      Attr positions:{x:+0.0,y:-5.0,z:+0.0}
      Attr normals:{x:+0.0,y:-1.0,z:+0.0}
      Attr uvs0:{x:+0.5,y:+1.0}
  Polygons (connected points as 'point.polygonPointIndex', borders as '|'):
    0: 3 points: 0.1, 1.0 |, 2.1
    1: 3 points: 0.2, 2.0 |, 3.1
    2: 3 points: 0.3, 3.0 |, 4.1
    3: 3 points: 0.4, 4.0 |, 5.1
    4: 3 points: 0.0, 5.0 |, 1.1
    5: 4 points: 1.2 |, 2.3, 7.2, 6.2
    6: 4 points: 2.2 |, 3.3, 8.2, 7.3
    7: 4 points: 3.2 |, 4.3, 9.2, 8.3
    8: 4 points: 4.2 |, 5.3, 10.2, 9.3
    9: 4 points: 5.2 |, 1.3, 6.1, 10.3
    10: 3 points: 6.3, 7.1, 11.1
    11: 3 points: 7.0, 8.1, 11.2
    12: 3 points: 8.0, 9.1, 11.3
    13: 3 points: 9.0, 10.1, 11.4
    14: 3 points: 10.0, 6.0, 11.0

NumGeometries:1
Mesh: pointCount: 12 polygonCount: 15 nbAttributeVectors: 24
  Points (adjacent polygons as 'polygon.neighborPolygonIndex', 
          borders as '|', closed wing starts as '<<',
          unshared attribute as value@attributeIndex ):
    0: 5 polygons:  <<4.0, 0.0, 1.0, 2.0, 3.0
      Attr positions:{x:+0.0,y:+5.0,z:+0.0}
      Attr normals:{x:+0.0,y:+1.0,z:+0.0}
      Attr uvs0:{x:+0.5,y:+0.0}
    1: 4 polygons:  |0.1, 4.2,  |5.0, 9.1
      Attr positions:{x:+4.330078,y:+2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:+0.5,z:+0.0}
      Attr uvs0:{x:+0.800048,y:+0.0}@1 {x:+0.200012,y:+0.333313}@16 {x:+1.0,y:+0.333313}@17 {x:+0.0,y:+0.333313}@22 
    2: 4 polygons:  |1.1, 0.2,  |6.0, 5.1
      Attr positions:{x:+1.338134,y:+2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:+0.823608}
      Attr uvs0:{x:+0.599975,y:+0.0}@12 {x:+1.0,y:+0.333313}@2 {x:+0.800048,y:+0.333313}@18 {x:+0.800048,y:+0.333313}@18 
    3: 4 polygons:  |2.1, 1.2,  |7.0, 6.1
      Attr positions:{x:-3.502929,y:+2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:+0.509033}
      Attr uvs0:{x:+0.400024,y:+0.0}@13 {x:+0.800048,y:+0.333313}@3 {x:+0.599975,y:+0.333313}@19 {x:+0.599975,y:+0.333313}@19 
    4: 4 polygons:  |3.1, 2.2,  |8.0, 7.1
      Attr positions:{x:-3.502929,y:+2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:-0.509033}
      Attr uvs0:{x:+0.200012,y:+0.0}@14 {x:+0.599975,y:+0.333313}@4 {x:+0.400024,y:+0.333313}@20 {x:+0.400024,y:+0.333313}@20 
    5: 4 polygons:  |4.1, 3.2,  |9.0, 8.1
      Attr positions:{x:+1.338134,y:+2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:-0.823608}
      Attr uvs0:{x:+0.0,y:+0.0}@15 {x:+0.400024,y:+0.333313}@5 {x:+0.200012,y:+0.333313}@21 {x:+0.200012,y:+0.333313}@21 
    6: 4 polygons:  <<14.1, 9.2, 5.3, 10.0
      Attr positions:{x:+4.330078,y:-2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:-0.5,z:+0.0}
      Attr uvs0:{x:+0.0,y:+0.666626}@23 {x:+0.0,y:+0.666626}@23 {x:+1.0,y:+0.666626}@6 {x:+1.0,y:+0.666626}@6 
    7: 4 polygons:  <<11.0, 10.1, 5.2, 6.3
      Attr positions:{x:+1.338134,y:-2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:+0.823608}
      Attr uvs0:{x:+0.800048,y:+0.666626}
    8: 4 polygons:  <<12.0, 11.1, 6.2, 7.3
      Attr positions:{x:-3.502929,y:-2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:+0.509033}
      Attr uvs0:{x:+0.599975,y:+0.666626}
    9: 4 polygons:  <<13.0, 12.1, 7.2, 8.3
      Attr positions:{x:-3.502929,y:-2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:-0.509033}
      Attr uvs0:{x:+0.400024,y:+0.666626}
    10: 4 polygons:  <<14.0, 13.1, 8.2, 9.3
      Attr positions:{x:+1.338134,y:-2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:-0.823608}
      Attr uvs0:{x:+0.200012,y:+0.666626}
    11: 5 polygons:  <<14.2, 10.2, 11.2, 12.2, 13.2
      Attr positions:{x:+0.0,y:-5.0,z:+0.0}
      Attr normals:{x:+0.0,y:-1.0,z:+0.0}
      Attr uvs0:{x:+0.5,y:+1.0}
  Polygons (connected points as 'point.polygonPointIndex', borders as '|'):
    0: 3 points: 0.1, 1.0 |, 2.1
    1: 3 points: 0.2, 2.0 |, 3.1
    2: 3 points: 0.3, 3.0 |, 4.1
    3: 3 points: 0.4, 4.0 |, 5.1
    4: 3 points: 0.0, 5.0 |, 1.1
    5: 4 points: 1.2 |, 2.3, 7.2, 6.2
    6: 4 points: 2.2 |, 3.3, 8.2, 7.3
    7: 4 points: 3.2 |, 4.3, 9.2, 8.3
    8: 4 points: 4.2 |, 5.3, 10.2, 9.3
    9: 4 points: 5.2 |, 1.3, 6.1, 10.3
    10: 3 points: 6.3, 7.1, 11.1
    11: 3 points: 7.0, 8.1, 11.2
    12: 3 points: 8.0, 9.1, 11.3
    13: 3 points: 9.0, 10.1, 11.4
    14: 3 points: 10.0, 6.0, 11.0

